packages/loot-core/src/server/main.ts

```diff
@@ -1670,14 +1670,11 @@ handlers['set-server-url'] = async function ({ url, validate = true }) {
 
     if (validate) {
       // Validate the server is running
-      const { error } = await runHandler(
-        handlers['subscribe-needs-bootstrap'],
-        {
-          url,
-        },
-      );
-      if (error) {
-        return { error };
+      const result = await runHandler(handlers['subscribe-needs-bootstrap'], {
+        url,
+      });
+      if ('error' in result) {
```

Comment:
This is what we have to do to type assert the union being returned from `handlers['subscribe-needs-bootstrap']`